qbft/spectest/tests/commit/valid_full_data.go

```diff
@@ -1,8 +1,34 @@
 package commit
 
-import "github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+import (
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
 
 // ValidFullData tests the signed commit with a valid full data field (H(full data) == root)
 func ValidFullData() tests.SpecTest {
-	panic("implement")
+	pre := testingutils.BaseInstance()
+	ks := testingutils.Testing4SharesSet()
+
+	msgs := []*qbft.SignedMessage{
+		testingutils.TestingProposalMessage(ks.Shares[1], 1),
+
+		testingutils.TestingPrepareMessage(ks.Shares[1], 1),
+		testingutils.TestingPrepareMessage(ks.Shares[2], 2),
+		testingutils.TestingPrepareMessage(ks.Shares[3], 3),
+
+		testingutils.TestingCommitMessageWithFullData(ks.Shares[1], 1, testingutils.TestingQBFTFullData),
+		testingutils.TestingCommitMessageWithFullData(ks.Shares[2], 2, testingutils.TestingQBFTFullData),
+		testingutils.TestingCommitMessageWithFullData(ks.Shares[3], 3, testingutils.TestingQBFTFullData),
```

Comment:
why do we need this? enough to have proposal +prepares, no?

Reply:
By the test desc:
> tests the signed commit with a valid full data field (H(full data) == root)

Isn't it supposed to process at least one commit message as described? I could make it process only one, though I included three of them.

Reply:
I see the comment saying:
> // ValidFullData tests signed decided with a valid full data field

so I thought you can just use the decided message... actually it should be able to terminate the instance w.o proposals and prepares, no?

Reply:
Wait. In this file I see

> // ValidFullData tests the signed COMMIT with a valid full data field (H(full data) == root)

Sure you see "decided"?

Reply:
Looked at wrong test